deeppavlov/core/layers/embeddings.py

```diff
@@ -0,0 +1,141 @@
+# Copyright 2017 Neural Networks and Deep Learning lab, MIPT
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Union, Sequence
+
+import tensorflow as tf
+from tensorflow.python.eager import context
+from tensorflow.python.keras.utils import tf_utils
+from tensorflow.python.framework import ops, tensor_shape
+from tensorflow.python.ops import embedding_ops
+
+
+class AdvancedEmbedding(tf.keras.layers.Layer):
+    """
+    Calculate token type ids and return the sum of subword-token, positional end token-type embeddings.
+
+    Args:
+        vocab_size: size of the token embedding vocabulary
+        token_type_vocab_size: the vocabulary size of `token_type_ids` (or `segment_ids`)
+        sep_token_index: index of separator-token used for calculating `token_type_ids` (or `segment_ids`)
+        output_dim: width of the word embeddings
+        use_one_hot_embeddings: if True, use one-hot method for word embeddings; if False, use
+            `tf.nn.embedding_lookup()`. One hot is better for TPUs.
+        max_len: maximum sequence length that might ever be used with this model. This can be longer than the sequence
+            length of input_tensor, but cannot be shorter.
+        initializer_range: stddev for embedding initialization range (TruncatedNormal initializer is used)
+        trainable_pos_embedding: whether the positional embedding matrix is trainable
+        **kwargs: keyword arguments for base Layer class
+
+    Note:
+        `token_type_ids` (or `segment_ids`) are calculated on the fly, and user of this layer is responsible
+        for token_type_vocab_size to be enough to deal with the inputs
+    """
+    # TODO: maybe break down this class into three distinct ones
+    def __init__(self,
+                 vocab_size: int = 119547,
```

Comment:
Remove defaults for vocab_size, max_len, token_type_vocab_size, sep_token_index and output_dim. They should be specified in BERT model directly.

Reply:
Again, we need to think about use cases. If they are 99% of the time limited to RuBERT + MultiBERT-base than these are reasonable defaults.

Reply:
Oh, yes. We cannot specify them from BERT model class as I thought before.
Add a comment where are they from.